### PR TITLE
Add Sin and Cos to chainerx

### PR DESCRIPTION
**Pull request by kshitij12345**
This PR references #6423 .

Changelog:
* Added `sin` and `cos` to chainerx routines. [ In the same fashion as `tanh`].
* Added tests in [math_test.cc](https://github.com/kshitij12345/chainer/blob/14b88f17bca44037c8b76e115dfb0d6ad0a5e197/chainerx_cc/chainerx/routines/math_test.cc#L1706-L1762) and [test_math.py](https://github.com/kshitij12345/chainer/blob/14b88f17bca44037c8b76e115dfb0d6ad0a5e197/tests/chainerx_tests/unit_tests/routines_tests/test_math.py#L1100-L1161).
* Updated to docs.

### CODE REVIEW

**Comment by asi1024** [comment]
Thanks for the PR! Could you fix also `docs/source/chainerx/reference/routines.rst`?

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py [hunk 1]
```diff
@@ -1097,6 +1097,70 @@ def forward_xp(self, inputs, xp):
...
+@op_utils.op_test(['native:0', 'cuda:0'])
+@pytest.mark.parametrize('input', [
+    numpy.asarray(0), numpy.asarray(-1), numpy.asarray(1), numpy.asarray(
+        10), numpy.full((), 2), numpy.full((0,), 2), numpy.full((2, 3), 2),
+    numpy.random.randn(2, 4), numpy.random.randn(4, 4),
+    numpy.random.randn(1, 4)
+])
+@pytest.mark.parametrize('contiguous', [None, 'C'])
+@pytest.mark.parametrize('in_dtype,out_dtype', _expected_dtypes_math_functions)
+class TestSin(op_utils.NumpyOpTest):
```
Comment (asi1024) [inline tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: How about using `UnaryMathTestBase` as the base class? It seems to reduce code duplication.

**Comment by asi1024** [comment]
LGTM except for comments!

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py (hunk 1 above)
Comment (kshitij12345) [reply tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: Makes sense. Will do that.

File: tests/chainerx_tests/unit_tests/routines_tests/test_math.py [hunk 2]
```diff
@@ -1098,67 +1098,51 @@ def forward_xp(self, inputs, xp):
...
-            self.skip_backward_test = True
-            self.skip_double_backward_test = True
-
-    def generate_inputs(self):
-        return self.input,
+@chainer.testing.parameterize(*(
+    # Special shapes
+    chainer.testing.product({
+        'shape': [(), (0,), (1,), (2, 0, 3), (1, 1, 1), (2, 3)],
+        'dtype': chainerx.testing.all_dtypes,
```
Comment (asi1024) [inline tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: Could you merge cuurent master branch and write as `'expected_dtypes': _expected_dtypes_math_functions,`?

Comment (kshitij12345) [reply tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: Sure on it.

Comment (kshitij12345) [reply tests/chainerx_tests/unit_tests/routines_tests/test_math.py]: I ll rebase on the current master instead of merge. Hope that is okay.

**Comment by asi1024** [comment]
LGTM. Jenkins, test this please.

**Comment by asi1024** [comment]
Jenkins failure is unrelated.

**Comment by asi1024** [comment]
LGTM!

**Pull request merged**